vendor/github.com/packer-community/winrmcp/winrmcp/winrmcp.go

```diff
@@ -42,7 +43,17 @@ func New(addr string, config *Config) (*Winrmcp, error) {
 		config = &Config{}
 	}
 
-	params := winrm.DefaultParameters()
+	connectTimeout := winrm.DefaultParameters.Timeout
+	if config.ConnectTimeout.Seconds() > 0 {
+		connectTimeout = iso8601.FormatDuration(config.ConnectTimeout)
```

Comment:
The `Timeout` you get from the `DefaultParameters` is not of the same nature of the timeout I added. The first one is used by the winrm server, the latter is for the client side when the server doesn't answer.
I'm not sure you can safely inherit the former from the latter.

If you want winrmcp to use this new connect timeout, you need to inject it in the `Endpoint` (a bit above in the code), not in the parameters.